File: settlement/room/main/furnisher/FurnisherItemTile.java
package settlement.room.main.furnisher;

import settlement.path.AVAILABILITY;
import settlement.room.sprite.RoomSprite;
import snake2d.util.map.MAP_BOOLEAN;
import snake2d.util.sets.INDEXED;

public class FurnisherItemTile implements INDEXED{

	private final int index;
	public final boolean canGoCandle;
	public final RoomSprite sprite;
	public final AVAILABILITY availability;
	public final boolean mustBeReachable;
	public boolean noWalls;
	private int data;

	public FurnisherItemTile(Furnisher p, boolean mustBeReachable, RoomSprite sprite, AVAILABILITY availability,	boolean canGoCandle) {
		this.index = p.tiles.add(this);
		this.canGoCandle = canGoCandle;
		this.sprite = sprite;
		this.availability = availability;
		this.mustBeReachable = mustBeReachable;
	}

	public FurnisherItemTile(Furnisher p, RoomSprite sprite, AVAILABILITY availability, boolean canGoCandle) {
		this(p, false, sprite, availability, canGoCandle);
	}

	public boolean isBlocker() {
		return availability.player < 0 || availability.from > 0;
	}
	
	public boolean isNotBlocker() {
		return !isBlocker() && availability.player <= AVAILABILITY.ROOM.player;
	}
	
	public CharSequence isPlacable(int tx, int ty, MAP_BOOLEAN roomIs, FurnisherItem it, int rx, int ry) {
		
		return null;
	}

	@Override
	public int index() {
		return index;
	}
	
	public FurnisherItemTile setData(int data) {
		this.data = data;
		return this;
	}
	
	public int data() {
		return data;
	}
	
	public RoomSprite sprite() {
		return sprite;
	}
	
}

File: LegacySource/settlement/room/main/furnisher/FurnisherItemTile.cs
using System;
using System.Collections.Generic;

namespace Settlement.Room.Main.Furnisher
{
    public class FurnisherItemTile : INDEXED
    {
        private readonly int index;
        public readonly bool CanGoCandle;
        public readonly RoomSprite Sprite;
        public readonly AVAILABILITY Availability;
        public readonly bool MustBeReachable;
        public bool NoWalls;
        private int data;

        public FurnisherItemTile(Furnisher p, bool mustBeReachable, RoomSprite sprite, AVAILABILITY availability, bool canGoCandle)
        {
            this.index = p.Tiles.Add(this);
            this.CanGoCandle = canGoCandle;
            this.Sprite = sprite;
            this.Availability = availability;
            this.MustBeReachable = mustBeReachable;
        }

        public FurnisherItemTile(Furnisher p, RoomSprite sprite, AVAILABILITY availability, bool canGoCandle)
            : this(p, false, sprite, availability, canGoCandle)
        {
        }

        public bool IsBlocker()
        {
            return Availability.Player < 0 || Availability.From > 0;
        }

        public bool IsNotBlocker()
        {
            return !IsBlocker() && Availability.Player <= AVAILABILITY.ROOM.Player;
        }

        public string IsPlacable(int tx, int ty, MAP_BOOLEAN roomIs, FurnisherItem it, int rx, int ry)
        {
            return null;
        }

        public int Index()
        {
            return index;
        }

        public FurnisherItemTile SetData(int data)
        {
            this.data = data;
            return this;
        }

        public int Data()
        {
            return data;
        }

        public RoomSprite Sprite()
        {
            return Sprite;
        }
    }
}